promql/parser/printer.go

```diff
@@ -15,14 +15,14 @@ package parser
 
 import (
 	"fmt"
-	"sort"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/promlog"
 	"strings"
-	"time"
-
-	"github.com/prometheus/common/model"
-	"github.com/prometheus/prometheus/pkg/labels"
 )
 
+var logger = promlog.New(&promlog.Config{})
```

Comment:
Shouldn't we inform the user in case of some error during the prettifying process?

Reply:
When called `String()`, I would expect any form of string representation and won't care if it's prettified, as long as the output is deterministic. Also, why would prettifying a valid expression error out?

Reply:
There should not be any error since the only errors possible is from the parser and the AST which are already checked well before entering the `.String()`. So, yes there will not be any errors as returns from any underlying function.